src/backend/aspen/app/views/tests/test_phylo_tree_view.py

```diff
@@ -1,46 +1,76 @@
 import json
 import random
+from typing import Collection, List, Mapping, Tuple, Union
 
 from aspen.app.views.phylo_trees import PHYLO_TREE_KEY
+from aspen.database.models import (
+    CanSee,
+    DataType,
+    Group,
+    PhyloTree,
+    Sample,
+    UploadedPathogenGenome,
+    User,
+)
 from aspen.test_infra.models.phylo_tree import phylorun_factory, phylotree_factory
 from aspen.test_infra.models.sample import sample_factory
 from aspen.test_infra.models.sequences import uploaded_pathogen_genome_factory
 from aspen.test_infra.models.usergroup import group_factory, user_factory
 
 
-def test_phylo_tree_view(
-    session,
-    app,
-    client,
-    n_samples=5,
-    n_trees=3,
-):
-    group = group_factory()
-    user = user_factory(group)
-    samples = [
+def make_sample_data(group: Group, n_samples: int) -> Collection[Sample]:
+    samples: Collection[Sample] = [
         sample_factory(
             group,
             public_identifier=f"public_identifier_{ix}",
             private_identifier=f"private_identifier_{ix}",
         )
         for ix in range(n_samples)
     ]
-    _ = [uploaded_pathogen_genome_factory(sample, accessions=()) for sample in samples]
+    return samples
+
 
-    # make up to 3 trees, each with a random sample of uploaded pathogen genomes.
-    trees = [
+def make_uploaded_pathogen_genomes(
+    samples: Collection[Sample],
+) -> Collection[UploadedPathogenGenome]:
+    return [
+        uploaded_pathogen_genome_factory(sample, accessions=()) for sample in samples
+    ]
+
+
+def make_trees(
+    group: Group, samples: Collection[Sample], n_trees: int
+) -> Collection[PhyloTree]:
+    # make up to n trees, each with a random sample of uploaded pathogen genomes.
+    return [
         phylotree_factory(
             phylorun_factory(group),
-            random.sample(samples, k=random.randint(0, len(samples))),
+            random.sample(samples, k=random.randint(0, len(samples))),  # type: ignore
```

Comment:
What happened here that is triggering the type checker?

Reply:
`aspen/app/views/tests/test_phylo_tree_view.py:48: error: Argument 1 to "sample" has incompatible type "Collection[Sample]"; expected "Union[Sequence[Any], AbstractSet[Any]]"`